client/src/helpers/APICalls/profile.ts

```diff
@@ -0,0 +1,15 @@
+import { FetchOptions } from '../../interface/FetchOptions';
+
+const getProfileDetails = async( profile_id: string ) => {
+    const fetchOptions: FetchOptions = {
+        method: 'GET',
+        credentials: 'include',
+      };
+      return await fetch(`/profiles/search?_id=${profile_id}`, fetchOptions)
+        .then((res) => res.json())
+        .catch(() => ({
+          error: { message: 'Unable to connect to server. Please try again' },
+        }));
```

Comment:
I would recommend against mixing `then/catch` and `async/await` syntax within the same function.

I would also suggest doing all sorts of response handling on the API layer (e.g. within `getProfileDetails`) and return a ready-to-use, consistent structure from API calls which simplifies UI layer, e.g.:

```
try {
  const resp = await fetch("...");
  const data = await resp.json();
  return { data: data.profiles[0] };
} catch (err) {
  return { error: err };
}
```
Once again, you're free to adopt these practices or not.

Reply:
sure thanks ugur